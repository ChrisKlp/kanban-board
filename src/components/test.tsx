import React from 'react';
import styled from 'styled-components';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

const Button = styled.button`
  color: red;
  font-size: 10px;
  line-height: 10px;
  border-radius: 50%;
`;

describe('style tests', () => {
  test('it renders correct styles for X', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', 'red');
  });
});

describe('other tests', () => {
  test('it renders correct markup for case X', () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
