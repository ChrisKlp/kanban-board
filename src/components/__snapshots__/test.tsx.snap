// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`other tests it renders correct markup for case X 1`] = `
.c0 {
  color: red;
  font-size: 10px;
  line-height: 10px;
  border-radius: 50%;
}

<button
  class="c0"
/>
`;

exports[`style tests it renders correct styles for X 1`] = `
.c0 {
  color: red;
  font-size: 10px;
  line-height: 10px;
  border-radius: 50%;
}

<button
  className="c0"
/>
`;
